Jenkinsfile

```diff
@@ -0,0 +1,61 @@
+node {
+    def app
+    def base_image = 'm4n/mojofy_frontend'
+    environment {
+        FRONTEND_TAG = 'latest'
+    }
+
+    stage('Clone repository') {
+        checkout scm
+    }
+
+
+    stage('Test image') {
+        def dockerfile = 'Dockerfile.test'
+        def testImage = docker.build("my-image:${env.BUILD_ID}", "-f ${dockerfile} ./")
+
+        testImage.inside('-e "NODE_ENV=production" -e BACKEND_API_URL=http://${DOCKER_MACHINE_IP}:3000') {
+            FRONTEND_TAG = sh (
+                script: "node -e \"console.log(require(\'./package.json\').version);\"",
+                returnStdout: true
+                ).trim()
+            sh "cd /app/src && ../node_modules/gulp/bin/gulp.js build"
+            try {
+                sh "cd /app/src && ../node_modules/karma/bin/karma start --watch false --single-run true"
+                sh "cp -r /app/src/report ${WORKSPACE}"
+            } catch(err) {
+                sh "echo TESTS FAILED"
+                step ([$class: 'JUnitResultArchiver', testResults: 'report/*.xml', healthScaleFactor: 1.0])
+                currentBuild.result = 'UNSTABLE'
+                throw err
+            }
+            sh 'echo "Tests passed"'
+            step ([$class: 'JUnitResultArchiver', testResults: 'report/*.xml', healthScaleFactor: 1.0])
+        }
+    }
+
+
+    stage('Build image') {
+        app = docker.build("${base_image}:${FRONTEND_TAG}")
+        echo "fronted tag is ${FRONTEND_TAG}"
+        echo "Built container image with frontend version ${env.BUILD_NUMBER}"
+    }
+
+//    stage ('Report') {
+//    //    step([$class: 'CheckStylePublisher', pattern: '**/checkstyle-result.xml'])
+//      unstash 'checkstyle-reports'
+//      step([$class: 'hudson.plugins.checkstyle.CheckStylePublisher', checkstyle: 'app/build/reports/checkstyle/checkstyle-result.xml'])
+//
+//      unstash 'junit-reports'
+//      step([$class: 'JUnitResultArchiver', testResults: '**/test-results/**/*.xml'])
+//    }
+
+
+    stage('Push image') {
+        docker.withRegistry("https://891817301160.dkr.ecr.us-east-2.amazonaws.com/m4n/mojofy_frontend", "ecr:us-east-2:m4n-aws") {
```

Comment:
Shall this registry be public? Will always be the same? I mean, shall it be an environment variable?

Reply:
public for now it isn't problem, as it's password protected, and password is managed by jenkins credentials

Not sure if always be the same, when we'll need to change it, we'll see what's the best approach :m